langchain-tool-retrieval.ipynb

```diff
@@ -474,401 +550,48 @@
e_per_share=26.271147540983605, shareholders_equity_per_share=26.271147540983605, interest_debt_per_share=104.39267759562841, enterprise_value=93198400000.0, price_to_sales_ratio=0.04553917050691244, pocf_ratio=0.5998138606382691, pfcf_ratio=-0.18673861465135322, pb_ratio=0.18499382230708752, ptb_ratio=0.18499382230708752, ev_to_sales=0.9544126984126984, enterprise_value_over_ebitda=7.061982844845876, ev_to_operating_cash_flow=12.5709352828509, ev_to_free_cash_flow=-3.913679215571, earnings_yield=0.19341563786008228, free_cash_flow_yield=-5.355078818952529, debt_to_equity=3.973662643886164, debt_to_assets=0.5500277261611157, net_debt_to_ebitda=6.725025005304156, current_ratio=0.3312362719620038, interest_coverage=0.0, income_quality=8.619695384257644, dividend_yield=0.3347500505970451, payout_ratio=1.730728985001744, sales_general_and_administrative_to_revenue=0.0, research_and_development_to_revenue=0.0, intangibles_to_total_assets=0.0, capex_to_operating_cash_flow=-4.212050500418139, capex_to_revenue=-0.3197880184331797, capex_to_depreciation=-9.804797638858362, stock_based_compensation_to_revenue=0.03870558115719406, graham_number=23.571906382212774, roic=0.05453134679235622, return_on_tangible_assets=0.0049527042940136256, graham_net_net=-149.34027322404373, working_capital=-58519100000.0, tangible_asset_value=24038100000.0, net_current_asset_value=-120640300000.0, invested_capital=3.973662643886164, average_receivables=55159900000.0, average_payables=11139550000.0, average_inventory=6966100000.0, days_sales_outstanding=13.220737327188939, days_payables_outstanding=50.9906274157384, days_of_inventory_on_hand=33.29467297231812, receivables_turnover=27.608142493638677, payables_turnover=7.158178247623242, inventory_turnover=10.962714675211513, roe=0.035780698141700054, capex_per_share=-34.128196721311475), FMPKeyMetricsData(symbol=F, market_cap=7707323598.24, pe_ratio=2.0097323593846155, date=1989-12-31, period=FY, calendar_year=1989, revenue_per_share=103.04032572102604, net_income_per_share=4.110000001457522, operating_cash_flow_per_share=10.207482402576838, free_cash_flow_per_share=-132.93197527530313, cash_per_share=5.810050328005651, book_value_per_share=25.21493038703841, tangible_book_value_per_share=25.39369127367286, shareholders_equity_per_share=25.21493038703841, interest_debt_per_share=91.532647229331, enterprise_value=87694223598.24, price_to_sales_ratio=0.08016279007466777, pocf_ratio=0.8092103100677202, pfcf_ratio=-0.06213704402491182, pb_ratio=0.32758369240812996, ptb_ratio=0.32758369240812996, ev_to_sales=0.9120953009773688, enterprise_value_over_ebitda=4.99374878127648, ev_to_operating_cash_flow=9.207225953933541, ev_to_free_cash_flow=-0.7069976708514765, earnings_yield=0.4975786926704022, free_cash_flow_yield=-16.09345947642895, debt_to_equity=3.6300971616555735, debt_to_assets=0.5308375177835248, net_debt_to_ebitda=4.554855131884652, current_ratio=1.9468094368875684, interest_coverage=0.0, income_quality=2.483572359843546, d
```

Comment:
Nit: no longer necessary because of Python 3.10+ (we can use built-in types directly as type hints)